libfreerdp/core/info.c

```diff
@@ -821,6 +820,9 @@ static void rdp_write_info_packet(rdpRdp* rdp, wStream* s)
 		}
 	}
 
+	/* excludes (!) the length of the mandatory null terminator */
+	cbAlternateShell = cbAlternateShell > 0 ? cbAlternateShell - 2 : 0;
```

Comment:
I´d change that to `cbAlternateShell = cbAlternateShell > 2 ? cbAlternateShell - 2 : cbAlternateShell;` to avoid a possible overflow